front/lib/resources/group_resource.ts

```diff
@@ -195,6 +195,70 @@ export class GroupResource extends BaseResource<GroupModel> {
     return group;
   }
 
+  /**
+   * Finds the specific editor groups associated with a set of agent configuration.
+   */
+  static async findEditorGroupsForAgents(
+    auth: Authenticator,
+    agent: LightAgentConfigurationType[]
+  ): Promise<Result<Record<string, GroupResource>, Error>> {
+    const owner = auth.getNonNullableWorkspace();
+
+    const groupAgents = await GroupAgentModel.findAll({
+      where: {
+        agentConfigurationId: agent.map((a) => a.id),
+        workspaceId: owner.id,
+      },
+      attributes: ["groupId", "agentConfigurationId"],
+    });
+
+    if (groupAgents.length === 0) {
+      return new Err(
+        new DustError(
+          "group_not_found",
+          "Editor group association not found for agent."
+        )
+      );
+    }
+
+    const groups = await GroupResource.fetchByIds(
+      auth,
+      groupAgents.map((ga) =>
+        GroupResource.modelIdToSId({
+          id: ga.groupId,
+          workspaceId: owner.id,
+        })
+      )
+    );
+
+    if (groups.isErr()) {
+      return groups;
```

Comment:
Means we return and display the raw error we have in the resource?

Reply:
yes we just return the error as is